src/flamegraph/flamegraph.js

```diff
@@ -1,21 +1,63 @@
-"use strict";
-var details, searchbtn, unzoombtn, matchedtxt, svg, searching;
+"use strict"
+var details, searchbtn, unzoombtn, matchedtxt, svg, searching, frames;
 function init(evt) {
     details = document.getElementById("details").firstChild;
     searchbtn = document.getElementById("search");
     unzoombtn = document.getElementById("unzoom");
     matchedtxt = document.getElementById("matched");
     svg = document.getElementsByTagName("svg")[0];
+    frames = document.getElementById("frames");
     searching = 0;
 
-    // use GET parameters to restore a flamegraphs state.
-    var params = get_params();
-    if (params.x && params.y)
-        zoom(find_group(document.querySelector('[x="' + params.x + '"][y="' + params.y + '"]')));
-    if (params.s)
-        search(params.s);
-}
+    // Use GET parameters to restore a flamegraph's state.
+    var restore_state = function() {
+        var params = get_params();
+        if (params.x && params.y)
+            zoom(find_group(document.querySelector('[x="' + params.x + '"][y="' + params.y + '"]')));
+        if (params.s)
+            search(params.s);
+    };
+
+    if (fluiddrawing) {
+        // Make width dynamic so the SVG fits its parent's width.
+        svg.removeAttribute("width");
+        // Edge requires us to have a viewBox that gets updated with size changes.
+        var isEdge = /Edge\/\d./i.test(navigator.userAgent);
+        if (!isEdge) {
+          svg.removeAttribute("viewBox");
+        }
+        var update_for_width_change = function() {
+            if (isEdge) {
+                svg.attributes.viewBox.value = "0 0 " + svg.width.baseVal.value + " " + svg.height.baseVal.value;
+            }
 
+            // Keep consistent padding on left and right of frames container.
+            frames.attributes.width.value = svg.width.baseVal.value - xpad * 2;
+
+            // Text truncation needs to be adjusted for the current width.
+            var el = frames.children;
+            for(var i = 0; i < el.length; i++) {
+                update_text(el[i]);
```

Comment:
I think we may be able to do _way_ better here by letting CSS take care of the text truncation using [`text-overflow: ellipsis`](https://css-tricks.com/almanac/properties/t/text-overflow/).

Reply:
Yeah, I can give that a try. I was just keeping it the way they were handling text.

Reply:
I think as we're moving to fluid layouts, we'll explicitly _not_ want to manually crop text, because it requires walking all the elements on every resize! Should also make the initial render faster I think.

Reply:
It doesn't look like `text-overflow` works for SVG text elements.

Reply:
Ah, that's a shame. I almost wonder whether it's worth just using `overflow: hidden` and not add the ellipses everywhere... Maybe with a very gentle fade on the right-hand side to indicate the crop?

Reply:
I can't get that to work either for SVG, unfortunately.